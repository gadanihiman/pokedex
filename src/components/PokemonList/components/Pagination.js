import React from 'react'
import { Icon } from 'antd';
import { PaginationWrapper } from '../styled';

const Pagination = ({
  onClickNext, onClickPrevious
}) => {
  return (
    <PaginationWrapper>
      <span onClick={onClickPrevious}>
        <Icon type="left" /> Previous  
      </span>
      <span onClick={onClickNext}>
        Next <Icon type="right" />
      </span>
    </PaginationWrapper>
  )
};

export default Pagination;

