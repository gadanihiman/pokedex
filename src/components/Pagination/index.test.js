import React from 'react';
import Pagination from '.';
import renderer from 'react-test-renderer';

describe('Pagination component', () => {
  test('Pagination should successfully rendered', () => {
    const component = renderer.create(
      <Pagination  />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
