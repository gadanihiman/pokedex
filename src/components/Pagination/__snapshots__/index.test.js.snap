// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component Pagination should successfully rendered 1`] = `
<div
  className="sc-AxiKw RHkGk"
>
  <span>
    <i
      aria-label="icon: left"
      className="anticon anticon-left"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="left"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
        />
      </svg>
    </i>
     Previous
  </span>
  <span>
    Next 
    <i
      aria-label="icon: right"
      className="anticon anticon-right"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="right"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
        />
      </svg>
    </i>
  </span>
</div>
`;
